src/uwtools/resources/rocoto.jsonschema

```diff
@@ -1,70 +1,251 @@
+          "$ref": "#/$defs/BoolLoop"
+        },
+        "^datadep(_.*)?$": {
+          "type": "object",
+          "properties": {
+            "attrs": {
+              "type": "object",
+              "properties": {
+                "age": {
+                  "$ref": "#/$defs/TimePattern"
                 },
-                "cycledefs": {
-                    "type": "object",
-                    "properties": {
-                        "groupname": {
-                            "type": "array",
-                            "items": {
-                                "$ref": "#/$defs/dateTimePattern"
-                            }, 
-                            "required": ["ref"]
-                        },
-                        "required": ["items"]
-                    }
+                "minsize": {
+                  "type": "string"
+                }
+              },
+              "additionalProperties": false
+            },
+            "text": {
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        }
+      },
+      "properties": {
+        "taskdep": {
+          "type": "object",
+          "properties": {
+            "task": {
+              "type": "string"
+            },
+            "cycle_offset": {
+              "$ref": "#/$defs/TimePattern"
+            },
+            "state": {
+              "type": "string",
+              "enum": [
+                "RUNNING",
+                "Running",
+                "running",
+                "SUCCEEDED",
+                "DEAD",
+                "Succeeded",
+                "Dead",
+                "succeeded",
+                "dead"
+              ]
+            }
+          },
+          "required": [
+            "task"
+          ]
+        },
+        "streq": {
+          "type": "object",
+          "properties": {
+            "left": {
+              "type": "string"
+            },
+            "right": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "left",
+            "right"
+          ]
+        },
+        "strneq": {
+          "type": "object",
+          "properties": {
+            "left": {
+              "type": "string"
+            },
+            "right": {
+              "type": "string"
+            }
+          },
+          "required": [
+            "left",
+            "right"
+          ],
+          "additionalProperties": false
+        },
+        "cycleexistdep": {
+          "type": "object",
+          "properties": {
+            "cycle_offset": {
+              "$ref": "#/$defs/TimePattern"
+            }
+          },
+          "additionalProperties": false
+        },
+        "taskvalid": {
+          "type": "object",
+          "properties": {
+            "task": {
+              "type": "string"
+            }
+          },
+          "additionalProperties": false
+        },
+        "metataskdep": {
```

Comment:
It's hard to tell from the indentation, but it's probably good to make sure the `metataskdep` section also has an additionalProperties tag applied.